### PR TITLE
Changed older versions button to links

### LINKED ISSUE #9811: Show download links instead of install buttons in all versions page
### Describe the problem and steps to reproduce it:

1. Install NoScript (possibly an older version).
2. Visit https://addons.mozilla.org/en-US/firefox/addon/noscript/versions/ and/or
  https://addons.mozilla.org/en-US/firefox/addon/noscript/
3. Try to install an older version, or at least find a "Download file" link (from #4910).

### What happened?
I am not able to install or download an older version. All buttons have changed to "Remove".

I cannot upgrade to the latest version via the addon listing because there is only a "Remove" button.

### What did you expect to happen?
It should be possible to install or download an older version.

### Anything else we should know?

Actually, it is possible to download, by right-clicking on the button and opening it in a new tab/window, because the button is actually a link. This is however not obvious and not a great user experience.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #9811 

This contains fixes from @sarthakkundra from #9973 and tests from @smitbarmase from #10098

### CODE REVIEW

File: src/amo/pages/AddonVersions/index.js [hunk 1]
```diff
@@ -175,6 +175,7 @@ export class AddonVersionsBase extends React.Component<InternalProps> {
                   headerText={i18n.gettext('Latest version')}
                   key="latestVersion"
                   version={latestVersion}
+                  showLinkInsteadOfButton={false}
```
Comment (bobsilverberg) [inline src/amo/pages/AddonVersions/index.js:178]: We shouldn't be specifying `false` for a boolean property, instead, we should just omit the property. I notice if we do that that Flow complains, so the definition of the `showLinkInsteadOfButton` property in `AddonVersionCard` will have to change to optional.

File: tests/unit/amo/components/TestInstallButtonWrapper.js [hunk 2]
```diff
@@ -328,23 +328,44 @@ describe(__filename, () => {
...
+    const showLinkInsteadOfButton = true;
+
+    const root = render({
+      _findInstallURL,
+      _getClientCompatibility,
+      version: createInternalVersionWithLang(fakeAddon.current_version),
+      showLinkInsteadOfButton,
+    });
+
+    expect(root.find('.InstallButtonWrapper-download')).toHaveLength(1);
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestInstallButtonWrapper.js:366]: We should also verify that the button is not displayed.

**Review by bobsilverberg — COMMENTED** [review]
This looks good, @smitbarmase. I have a comment about the code and a comment about one of the tests. I also noticed a behaviour that I think is not what we want: when viewing the versions page using a non-Firefox browser, we still see "download Firefox" buttons for each version, whereas I think we should only have one for the most recent version. All the extra buttons are kind of ugly.

Do you think you can address all of these issues? If so, it's probably best for you to take the patches from this PR and add them to a new one of your own so it can be your pull request once again, as opposed to my pull request.

**Comment by smitbarmase** [comment]
Yes. I'll look up to that @bobsilverberg

**Comment by smitbarmase** [comment]
Fixed those issues in #10237

**Comment by bobsilverberg** [comment]
Closing this as it has been replaced by https://github.com/mozilla/addons-frontend/pull/10237

**Pull request closed without merging**